### DISCUSSION TITLE
Sharing loaders across routes

### DISCUSSION THREAD

**Discussion by JUNIORCO** [remix-run/remix#7756] [Q&A] [answered]
```
.
├── routes/
│   ├── auth.sign-in/
│   │   └── route.tsx
│   ├── auth.sign-up/
│   │   └── route.tsx
│   └── _index.tsx
└── root.tsx
```

In the `sign-in` and `sign-up` routes I have the same `loader` function. Obviously I can abstract the loader into a util file, but was wondering if there's a way to do it using the folder structure.

**Comment by sergiodxa** [comment, marked as answer]
You will need to re-export the loader
```ts
export { loader } from "~/routes/auth.sign-in/route"
```
Or, my recommendation, move the shared logic somewhere else and just call a function from each route loader.

**Reply by JUNIORCO** [reply]
Hmm... Say I have `/auth/...` routes and `/app/...` routes. I'll have to export a loader auth-guard for each route under app? I get that I can move the shared logic but I would have expected some neat folder-structure solution

**Reply by sergiodxa** [reply]
Ahh you don't want to re-use the loader, you want to re-use the guard.

I recommend you to move that logic somewhere else and call it on each loader. Until #7642 it's implemented you will need to call your guard on every route because each loader run individually so you can't expect a parent loader to protect a child loader, think about loaders and actions as individual API endpoints.
